docs: add Rsdoctor documents (#7322)
web-infra-dev/rspack: website/docs/en/contribute/development/profiling.md

speedscope CPU.20230522.154658.14577.0.001.cpuprofile
```

### Rsdoctor Timeline

> If we want to analyze the time consumption of loaders and plugins or the compilation behavior of loaders, we can use Rsdoctor to view:

Reviewer: ```suggestion
If you need to analyze the time cost of loaders and plugins, or the compilation behavior of loaders, you can use Rsdoctor to view:
```

Author: This is because is always use 'we' in this document.

![image](https://github.com/web-infra-dev/rsdoctor/assets/18437716/eea845fa-f19f-49d4-9bc6-6795688e6a33)

> To enable `RsdoctorRspackPlugin`, you can refer to the documentation: [Use Rsdoctor](/guide/optimization/use-rsdoctor).

Reviewer: ```suggestion
To learn how to enable `RsdoctorRspackPlugin`, you can refer to [Use Rsdoctor](/guide/optimization/use-rsdoctor).
```

- The `RsdoctorRspackPlugin` requires the `features.loader` and `features.plugins` parameters to be set to `true`. By default, `features.loader` and `features.plugins` are enabled. Refer to the parameter documentation [Rsdoctor options](https://rsdoctor.dev/config/options/options#features).
- Usage documentation:
  - For Loader Analysis, refer to [Loader Timeline](https://rsdoctor.dev/guide/usage/loaders-timeline) and [Loader Details](https://rsdoctor.dev/guide/usage/loaders-analysis).

>   - For Plugins Analysis, refer to [Plugin Analysis](https://rsdoctor.dev/guide/usage/plugins-analysis).

Reviewer: This part can be moved to `Use Rsdoctor`
